integration/container/rename_test.go

```diff
@@ -86,3 +91,195 @@ func TestRenameLinkedContainer(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, []string{"/a0:/b0/a0"}, cntJSON.HostConfig.Links)
 }
+
+func TestRenameStoppedContainer(t *testing.T) {
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := request.NewAPIClient(t)
+
+	oldName := "first_name"
+	cID := runSimpleContainer(ctx, t, client, oldName, func(config *container.Config, hostConfig *container.HostConfig, networkingConfig *network.NetworkingConfig) {
+		config.Cmd = []string{"sh"}
+	})
+	poll.WaitOn(t, containerIsInState(ctx, client, cID, "exited"), poll.WithDelay(100*time.Millisecond))
+
+	inspect, err := client.ContainerInspect(ctx, cID)
+	require.NoError(t, err)
+	assert.Equal(t, inspect.Name, "/"+oldName)
+
+	newName := "new_name" + stringid.GenerateNonCryptoID()
+	err = client.ContainerRename(ctx, oldName, newName)
+	require.NoError(t, err)
+
+	inspect, err = client.ContainerInspect(ctx, cID)
+	require.NoError(t, err)
+	assert.Equal(t, inspect.Name, "/"+newName)
+}
+
+func TestRenameRunningContainer(t *testing.T) {
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := request.NewAPIClient(t)
+
+	oldName := "first_name"
+	cID := runSimpleContainer(ctx, t, client, oldName)
+	poll.WaitOn(t, containerIsInState(ctx, client, cID, "running"), poll.WithDelay(100*time.Millisecond))
+
+	inspect, err := client.ContainerInspect(ctx, cID)
+	require.NoError(t, err)
+	assert.Equal(t, inspect.Name, "/"+oldName)
+
+	newName := "new_name" + stringid.GenerateNonCryptoID()
+	err = client.ContainerRename(ctx, oldName, newName)
+	require.NoError(t, err)
+
+	inspect, err = client.ContainerInspect(ctx, cID)
+	require.NoError(t, err)
+	assert.Equal(t, inspect.Name, "/"+newName)
+}
+
+func TestRenameRunningContainerAndReuse(t *testing.T) {
```

Comment:
I don't think this test is necessary. It's basically identical to the previous `TestRenameRunningContainer`.

If we want to test "old name was removed" I think that could be done with an inspect and checking for "not found" which would make the test more obvious. 

Either way I think this is the same case, so these tests should be combined and one removed.

Reply:
Thanks @dnephin. Done. The tests of TestRenameRunningContainer/TestRenameRunningContainerAndReuse/TestRenameCheckNames have been merged.